src/mem/mod.rs

```diff
@@ -69,6 +69,7 @@ pub unsafe extern "C" fn bcmp(s1: *const u8, s2: *const u8, n: usize) -> i32 {
 }
 
 // `bytes` must be a multiple of `mem::size_of::<T>()`
+#[cfg_attr(target_has_atomic_load_store = "0", allow(dead_code))]
```

Comment:
Are you sure this actually works? We never generate "0" as a value for `target_has_atomic_load_store`.

Also `memcpy_element_unordered_atomic` should almost always be supported. It is a bug with the riscv target if it doesn't expose the proper `target_has_atomic_load_store` attributes.

On RISC-V normal loads and stores are guaranteed to be atomic, even if the A extension is not supported. It's just that RMW operations are not available.

Reply:
Hmm, I thought this (https://github.com/rust-lang/rust/pull/66548) created `target_has_atomic_load_store = "0"`. Then does just `#[allow(dead_code)]` work here?

Reply:
I see, the RISC-V LLVM target behaves differently from the ARM one in this regards. In any case, you should be able to just change the condition to `not(target_has_atomic_load_store = "8")`.

Reply:
Updated it, thanks!